src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx

```diff
@@ -142,7 +142,7 @@ namespace xclhwemhal2 {
           if (count >= size) break;
           if (map->m_debug_ip_data[i].m_type == type) {
             if (baseAddress) baseAddress[count] = map->m_debug_ip_data[i].m_base_address;
-            if (portNames)   portNames[count]   = (char*)map->m_debug_ip_data[i].m_name;
+            if (portNames)   portNames[count]   = map->m_debug_ip_data[i].m_name;
```

Comment:
is m_name guaranteed to be null-terminated?

Reply:
When the binary image of debug_ip_data is created in "XclBinData::createDebugIPLayoutBinaryImage", the entire debug_ip_data is first initialized with 0.
Then while writing the content of "m_name", the input string length is confirmed to be less than "m_name" size and then the input string along with its terminating null character is copied into m_name like this 
    memcpy( debugIpDataHdr.m_name, sm_name.c_str(), sm_name.length() + 1);

Also, while reading the debug_ip_data, the entire structure is read altogether and so, the terminating null will also be read in.

Reply:
I guess my point is that this is a lot of assumptions, none of which can be deduced from looking at this code :-) I think Fortify will keep complaining, and I'd say rightfully so.  You should consider using the std::string::assign function to copy the char array content.

Reply:
Okay, then. I will update it.
Thanks!